apps/server/src/shared/infra/identity-management/keycloak-identity-management.spec.ts

```diff
@@ -0,0 +1,255 @@
+			providers: [
+				{ provide: 'IdentityManagement', useClass: KeycloakIdentityManagement },
+				{
+					provide: 'KeycloakAdminClient',
+					useValue: {
+						auth: jest.fn(),
+						setConfig: jest.fn(),
+						users: {
+							create: jest.fn(),
+							find: jest.fn(),
+							findOne: jest.fn(),
+							update: jest.fn(),
+							resetPassword: jest.fn(),
+							del: jest.fn(),
+						},
+					},
+				},
+				{
+					provide: 'KeycloakSettings',
+					useValue: 'empty',
+				},
+			],
+		}).compile();
+		idm = module.get<IdentityManagement>(IdentityManagement);
+		kcAdminClient = module.get<KeycloakAdminClient>('KeycloakAdminClient');
+	});
+
+	it('should be defined', () => {
+		expect(idm).toBeDefined();
+	});
+
+	describe('createAccount', () => {
+		it('should allow to create an account', async () => {
+			const accountId = 'user-1-id';
+			const createUserMock = jest.spyOn(kcAdminClient.users, 'create').mockResolvedValue({ id: accountId });
+			const resetPasswordMock = jest.spyOn(kcAdminClient.users, 'resetPassword');
+			const testAccount = { userName: 'test', email: 'test@test.test' };
+			const testAccountPassword = 'test';
+
+			const ret = await idm.createAccount(testAccount, testAccountPassword);
+
+			expect(ret).not.toBeNull();
+			expect(ret).toBe(accountId);
+			expect(createUserMock).toBeCalledWith(
+				expect.objectContaining({ username: testAccount.userName, email: testAccount.email })
+			);
+			expect(resetPasswordMock).toBeCalledWith(
+				expect.objectContaining({
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+					credential: expect.objectContaining({ value: testAccountPassword }),
+				})
+			);
+		});
+
+		it('should reject if account create fails', async () => {
+			jest.spyOn(kcAdminClient.users, 'create').mockResolvedValueOnce({ id: 'accountId' }).mockRejectedValue('error');
+			const resetPasswordMock = jest.spyOn(kcAdminClient.users, 'resetPassword').mockRejectedValue('error');
+			const deleteAccountMock = jest.spyOn(kcAdminClient.users, 'del');
+
+			const testAccount = { userName: 'test', email: 'test@test.test' };
+			const testAccountPassword = 'test';
+
+			await expect(idm.createAccount(testAccount, testAccountPassword)).rejects.toBeTruthy();
+			expect(resetPasswordMock).toHaveBeenCalled();
+			expect(deleteAccountMock).toHaveBeenCalled();
+			resetPasswordMock.mockClear();
+			deleteAccountMock.mockClear();
+
+			await expect(idm.createAccount(testAccount, testAccountPassword)).rejects.toBeTruthy();
+			expect(resetPasswordMock).not.toHaveBeenCalled();
+			expect(deleteAccountMock).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('findAccountById', () => {
+		it('should find an existing account', async () => {
+			const accountId = 'user-1-id';
+			const accountUserName = 'user-1';
+			const accountEmail = 'user@mail';
+			const accountFirstName = 'user';
+			const accountLastName = '1';
+
+			jest.spyOn(kcAdminClient.users, 'findOne').mockResolvedValue({
```

Comment:
If we used `createMock` the `jest.spyOn` wouldn't be necessary, plus the mock would be type-safe.

Reply:
@uidp As mentioned above, do you now of a work around to mock nested classes? Anyway, the test focuses on the class as a facade towards the KeycloakAdminClient rather then testing or expecting KeycloakAdminClient specifcs. Nevertheless, with deep mocking the test code can surely be compacted :)